pegasus/config/puma.rb

```diff
@@ -25,9 +26,48 @@
   PEGASUS_DB.disconnect
   DASHBOARD_DB.disconnect
   Cdo::AppServerMetrics.instance&.spawn_reporting_task if defined?(Cdo::AppServerMetrics)
+
+  # Control automated restarts of web application server processes via Gatekeeper.
+  # NOTE: before_fork runs on the parent puma process, so complete restart of the web application services on all
+  # front end instances is required for a change of this Gatekeeper flag to take effect:
+  #   sudo service dashboard upgrade && sudo service pegasus upgrade
+  if Gatekeeper.allows('enableWebServiceProcessRollingRestart')
+    require 'puma_worker_killer'
+
+    restart_period = DCDO.get("web_service_process_restart_period", 12 * 3600) # default to 12 hours
+    PumaWorkerKiller.enable_rolling_restart(restart_period)
+  end
+end
+
+on_worker_shutdown do
+  # This method posts metrics asynchronously to AWS, so this metric is unlikely to be published before worker shutdown.
```

Comment:
I think we should either figure out how to consistently publish this metric, or just remove this. Having the code here if it doesn't work is confusing, and if it works sometimes it could be a misleading metric.

Do we already have general metrics on the number of running workers or anything like that? That might be more useful, anyway.

Reply:
Good point.  I'll try to change this so we can publish total number of processes running on the instance on a regular schedule, or at least anytime an event (such as process startup) occurs.